### PR TITLE
Forest twisted hallway chest matches contents

### PR DESCRIPTION
**Pull request by SlyryD**
- When the forest hallway actor draws the chest on the wall, create a dummy chest actor to get the chest override and use the corresponding textures
- Scale the base and lid, and translate the lid so it appears on the base when open and closed
- Add EnBox struct (matches vanilla) and Chest struct (contains additional rando fields) to make accessing chest actor fields a little nicer
- Support invisible chests setting so the chest is visible when it is open or lens is active
- Refactor chest type constants into a ChestType enum that contains all possible values
- Refactor Matrix functions into sys_matrix.h

Chest Textures + Sizes
image
image
image
image
image
image
image
image
image
image

Chest Textures
image
image

Invisible Chests
image
image
image

### CODE REVIEW

File: ASM/c/chests.c [hunk 1]
```diff
@@ -41,87 +43,157 @@ void get_chest_override(z64_actor_t *actor) {
...
+        set_chest_texture(gfx, chest_type, opa_ptr);
+        gSPDisplayList((*opa_ptr)++, 0x060010C0);
+    }
+    else
+    {
+        gSPDisplayList((*opa_ptr)++, 0x06001678);
+    }
+}
 
+void draw_chest(z64_game_t *game, int32_t part, void *unk, void *unk2, z64_actor_t *actor, Gfx **opa_ptr)
```
Comment (SlyryD) [inline ASM/c/chests.c:145]: FYI refactored this function to make it almost identical to the original source code. Also split draw_chest_base and draw_chest_lid into separate functions so they can be called from forest hallway code separately.

**Comment by r0bd0g** [comment]
You need to support the invisible chests setting, if you haven't.

**Comment by SlyryD** [comment]
> You need to support the invisible chests setting, if you haven't.

In progress. Wrote most of it already.

**Comment by SlyryD** [comment]
> You need to support the invisible chests setting, if you haven't.

Done

**Comment by r0bd0g** [comment]
Are invisible chests always visible after they're open? They are, I think? Does this do that?

**Comment by SlyryD** [comment]
> Are invisible chests always visible after they're open? They are, I think? Does this do that?

Good point, I'll look into it.

**Comment by SlyryD** [comment]
> Are invisible chests always visible after they're open? They are, I think? Does this do that?

Fixed

**Pull request merged**